File: sort.py
import time
time_tests=[0]*100
for h in range(5,6):
    n=10**h
    print(h)
    for g in range(5,6):
        filename='test_'+str(n)+'_'+str(g)+'.txt'
        file=open(filename)
        arr=[int(file.readline()) for _ in range(n)]
        #ввод данных
        start=time.perf_counter()
        #начало работы 
        for i in range(1,n):
            left=0
            right=i
            while (right-left>1):
                q=(right+left)//2
                if arr[i]<=arr[q]:
                    right=q
                else:
                    left=q
            q=arr[i]
            arr.pop(i)
            arr.insert(left+1,q)
            if arr[0]>arr[1]:
                arr[0],arr[1]=arr[1],arr[0]
        
            #конец работы 
        print(arr)
        end=time.perf_counter()
        print(end-start)
        time_tests[g-1]=end-start
    print("mean=",sum(time_tests)/5)
        
    
         

File: sort.cpp
#include<iostream>
#include<string>
#include<fstream>
#include<vector>
#include<ctime>
using namespace std;
typedef long long ll;
typedef long double ld;
int main(){
    ld time_tests;
    for (ll h=8;h<9;h++){
        cout<<h<<endl;
        ll n=1;
        time_tests=0;
        for (ll i=0;i<h;i++){
            n*=10;
        }
        for (ll g=1;g<2;g++){

            string filename="test_"+to_string(n)+"_"+to_string(g)+".txt";
            ifstream file;
            file.open(filename);
            vector<ll>arr(n);
            ll a;
            for (ll i=0;i<n;i++){
                file>>a;
                arr[i]=a;
            }
            //ввод данныч

            ld start=clock();
            //начало работы
            for (int i=0;i<n;i++){
                for (int j=i;i>0;j--){
                    if (arr[j]<arr[j-1]){
                        a=arr[j];
                        arr[j]=arr[j-1];
                        arr[j-1]=a;
                    }else{
                        break;
                    }
                }
            }

    //конец работы
            ld send=clock();
            time_tests+=send-start;
            file.close();
            cout<<(send-start)/1000.0<<endl;
        }
        cout<<"mean="<<(time_tests)/1000.0<<endl;
    }
    return 0;
}
